orchestra/bots/sanitybot.py

```diff
@@ -0,0 +1,92 @@
+from django.db.models import Max
+from django.utils import timezone
+from pydoc import locate
+
+from orchestra.core.errors import SanityBotError
+from orchestra.models import Project
+from orchestra.models import SanityCheck
+from orchestra.models import WorkflowVersion
+from orchestra.utils.notifications import message_experts_slack_group
+from orchestra.utils.project_properties import incomplete_projects
+
+
+def _workflow_versions_with_sanity_checks():
+    # TODO(marcua): make more specific.
+    return WorkflowVersion.objects.all()
+
+
+def _handle(project, sanity_check, handler):
+    handler_type = handler.get('type')
+    handler_message = handler.get('message')
+    handler_steps = handler.get('steps')
+    if (handler_type != 'slack_project_channel' or
+            not handler_message or not handler_steps):
+        raise SanityBotError('Invalid handler: {}'.format(handler))
+    tasks = (
+        task for task in project.tasks.all()
+        if task.step.slug in handler_steps)
+    usernames = {
+        assignment.worker.formatted_slack_username()
+        for task in tasks
+        for assignment in task.assignments.all()
+        if assignment and assignment.worker
+    }
+    message = '{}: {}'.format(' '.join(usernames), handler_message)
+    message_experts_slack_group(
+        project.slack_group_id, message)
+
+
+def _filter_checks(project, checks, check_configurations):
+    latest_check_creation = (SanityCheck.objects
+                             .filter(project=project)
+                             .values('check_slug')
+                             .annotate(max_created_at=Max('created_at')))
+    latest_check_creation = {
+        check['check_slug']: check['max_created_at']
+        for check in latest_check_creation}
+    for check in checks:
+        max_created_at = latest_check_creation.get(check.check_slug)
```

Comment:
What is the different between check['check_slug'] and check.check_slug?

Reply:
`check` is of type `SanityCheck` (we both miss typed languages :)), which has a property called `check_slug`

`latest_check_creation` is a dictionary, which has keys defined by the set of distinct `check_slug`